### DISCUSSION TITLE
Yaml cuts off code on safe

### DISCUSSION THREAD

**Discussion by madflow** [LazyVim/LazyVim#1303] [Q&A] [answered]
When I edit `yaml` files with     

`{ import = "lazyvim.plugins.extras.formatting.prettier" },`

enabled - sometimes the code is cut-off at certain chars in the editor.

I understand, that this is most likely not related to LazyVim - but I have to start somewhere.

* wget https://raw.githubusercontent.com/goreleaser/goreleaser/main/.goreleaser.yaml

Note the end of the file:

```
tail -3 .goreleaser.yaml

  discord:
    enabled: true
    message_template: "GoReleaser {{ .Tag }} is out! Check it out: https://github.com/goreleaser/goreleaser/releases/tag/{{ .Tag }}"

```
* nvim .goreleaser.yaml

* When I jump around in the file - and save it at some point - the code is cutoff afterwards.

![Kapture 2023-08-07 at 19 08 24](https://github.com/LazyVim/LazyVim/assets/183248/d6aea3b7-9890-4173-bfb8-34accc7a5769)

* When I disable `{ import = "lazyvim.plugins.extras.formatting.prettier" },` - I cannot reproduce this anymore.
* Just running `prettier .goreleaser.yaml` works fine.
* So "probably" an issue with the prettierd formatting? 
* Can anyone else repdroduce this?

```
 Detected filetype:   yaml
 
 3 client(s) attached to this buffer: 
 
 Client: null-ls (id: 1, bufnr: [1])
 	filetypes:       fish, luau, lua, sh, markdown, handlebars, html, markdown.mdx, vue, json, less, typescriptreact, scss, jsonc, graphql, typescript, javascriptreact, yaml, css, javascript, dockerfile, go, gitrebase, php, zsh, csh, bash, ksh, python
 	autostart:       false
 	root directory:  /private/tmp/yaml
 	cmd:             <function>
 
 Client: yamlls (id: 2, bufnr: [1])
 	filetypes:       yaml, yaml.docker-compose
 	autostart:       true
 	root directory:  Running in single file mode.
 	cmd:             /Users/reiss/.local/share/nvim/mason/bin/yaml-language-server --stdio
 
 Client: docker_compose_language_service (id: 3, bufnr: [1])
 	filetypes:       yaml
 	autostart:       true
 	root directory:  Running in single file mode.
 	cmd:             /Users/reiss/.local/share/nvim/mason/bin/docker-compose-langserver --stdio
 
  
 Configured servers list: cssls, gopls, astro, ansiblels, tsserver, html, dockerls, taplo, docker_compose_language_service, eslint, rust_analyzer, yamlls, lua_ls, bashls, marksman, vuels, prismals, jsonls, intelephense, helm_ls
```

```
	null-ls
	https://github.com/jose-elias-alvarez/null-ls.nvim
	
	Logging
	* current level: warn
	* path: /Users/reiss/.cache/nvim/null-ls.log
	
	Active source(s)
	* name: prettierd
	* filetypes: markdown | handlebars | html | markdown.mdx | vue | json | less | typescriptreact | scss | jsonc | graphql | typescript | javascriptreact | yaml | css | javascript
	* methods: formatting | range_formatting
	
	* name: gitsigns
	* filetypes: *
	* methods: code_action
	
	
	Supported source(s)
	* formatting: prettier | prettier_d_slim | prettier_eslint | prettierd | yamlfix | yamlfmt | yq
	* diagnostics: actionlint | cfn_lint | spectral | vacuum | yamllint
```

**Comment by dpetka2001** [comment, marked as answer]
Please see this [comment](https://github.com/LazyVim/LazyVim/discussions/739#discussioncomment-6651586) in the issue. It has a link to a discussion in `null-ls` repo and it seems this is due to combination of `prettierd` and `null-ls`. The solution mentioned was to configure `prettier` instead of `prettierd` in your own spec.
